### PR TITLE
NN-2484 convert the rest of the tests to junit 5

### PR DESCRIPTION
**Pull request by JamieDraperUK**

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 version: "3"
 services:
   oauth:
-    image: mojdigitalstudio/nomis-oauth2-server:latest
+  image: mojdigitalstudio/nomis-oauth2-server:latest
```
Comment (brightonsbox) [inline docker-compose.yaml]: Is this right?

Comment (JamieDraperUK) [reply docker-compose.yaml]: Just automatic formatting done by my IDE. I was messing around with it during my testing, removing bits etc. The file is still valid though.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/whereabouts/services/LocationGroupServiceSelectorTest.kt [hunk 2]
```diff
@@ -5,12 +5,10 @@ import com.nhaarman.mockito_kotlin.verify
...
 import com.nhaarman.mockito_kotlin.whenever
 import org.assertj.core.api.Assertions.assertThat
-import org.junit.Test
-import org.junit.runner.RunWith
-import org.mockito.junit.MockitoJUnitRunner.StrictStubs
+import org.junit.jupiter.api.Test
 import uk.gov.justice.digital.hmpps.whereabouts.model.LocationGroup
 
-@RunWith(StrictStubs::class)
+//@RunWith(StrictStubs::class)
```
Comment (brightonsbox) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/whereabouts/services/LocationGroupServiceSelectorTest.kt]: @ExtendWith(MockitoExtension.class)
@MockitoSettings(strictness = Strictness.STRICT_STUBS) ?

Comment (JamieDraperUK) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/whereabouts/services/LocationGroupServiceSelectorTest.kt]: Ah good spot, forgot to remove the commented out code.  I think all stubs should be strict so will look at adding this to all tests in the future. Or even better look for a way to make them all strict by default.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/whereabouts/services/AttendanceStatisticsTest.kt [hunk 3]
```diff
@@ -1,27 +1,25 @@
...
-import org.mockito.junit.MockitoJUnitRunner
+
+import com.nhaarman.mockito_kotlin.mock
 import uk.gov.justice.digital.hmpps.whereabouts.model.AbsentReason
 import uk.gov.justice.digital.hmpps.whereabouts.model.Attendance
 import uk.gov.justice.digital.hmpps.whereabouts.model.TimePeriod
 import uk.gov.justice.digital.hmpps.whereabouts.repository.AttendanceRepository
 import java.time.LocalDate
 
-@RunWith(MockitoJUnitRunner::class)
```
Comment (JamieDraperUK) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/whereabouts/services/AttendanceStatisticsTest.kt]: Not needed.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/whereabouts/config/ClientTrackingTelemetryModuleTest.kt [hunk 4]
```diff
@@ -15,12 +15,12 @@ import org.springframework.mock.web.MockHttpServletRequest
...
 import org.springframework.test.context.ActiveProfiles
 import org.springframework.test.context.ContextConfiguration
-import org.springframework.test.context.junit4.SpringRunner
+import org.springframework.test.context.junit.jupiter.SpringExtension
 import uk.gov.justice.digital.hmpps.whereabouts.utils.JwtAuthenticationHelper
 import uk.gov.justice.digital.hmpps.whereabouts.utils.JwtParameters
 import java.time.Duration
 
-@RunWith(SpringRunner::class)
+@ExtendWith(SpringExtension::class)
```
Comment (petergphillips) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/whereabouts/config/ClientTrackingTelemetryModuleTest.kt]: unnecessary with junit 5

File: src/test/kotlin/uk/gov/justice/digital/hmpps/whereabouts/integration/PingEndpointIntegrationTest.kt [hunk 5]
```diff
@@ -1,17 +1,17 @@
...
 import org.springframework.boot.test.context.SpringBootTest
 import org.springframework.boot.test.web.client.TestRestTemplate
 import org.springframework.http.HttpStatus
 import org.springframework.test.context.ActiveProfiles
-import org.springframework.test.context.junit4.SpringRunner
+import org.springframework.test.context.junit.jupiter.SpringExtension
 import org.springframework.util.MimeType
 
-@RunWith(SpringRunner::class)
+@ExtendWith(SpringExtension::class)
```
Comment (petergphillips) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/whereabouts/integration/PingEndpointIntegrationTest.kt]: again unncessary

**Pull request merged**